src/code.cloudfoundry.org/silk/controller/leaser/cidrpool_test.go

```diff
@@ -157,20 +258,22 @@ var _ = Describe("CIDRPool", func() {
 				for result := range results {
 					_, subnet, err := net.ParseCIDR(result)
 					Expect(err).NotTo(HaveOccurred())
-					Expect(network.Contains(subnet.IP)).To(BeTrue())
+
+					isInNetwork0 := networks[0].Contains(subnet.IP)
+					Expect(isInNetwork0).To(BeTrue())
 					Expect(subnet.Mask).To(Equal(net.IPMask{255, 255, 255, 255}))
-					// 10.255.0.0 should never be allocated? Maybe?
-					Expect(subnet.IP.To4()).NotTo(Equal(network.IP.To4()))
+					//The first IP in the lease should never be allocated
+					Expect(subnet.IP.To4()).NotTo(Equal(networks[0].IP.To4()))
 				}
 			})
 		})
 
 		Context("when no subnets are available", func() {
 			It("returns an empty string", func() {
-				subnetRange := "10.255.0.0/16"
+				subnetRange := []string{"10.255.0.0/16", "10.250.0.0/16"}
 				cidrPool := leaser.NewCIDRPool(subnetRange, 24)
 				var taken []string
-				for i := 0; i < 255; i++ {
+				for i := 0; i < 510; i++ {
 					s := cidrPool.GetAvailableSingleIP(taken)
 					taken = append(taken, s)
```

Comment:
Same question about `taken`

Reply:
scratch that, but same question about restructuring this test as i had on the GetAvailableBlock one